### PR TITLE
Prepare for nightly CI

### PR DESCRIPTION
**Pull request by adinapoli-mndc**
This PR re-organises the benchmarks to allow for nightly CI.

@MeBrei I did some minor tweaks to your original benchmarking code, i.e. I have factored out "fixtures" code into a separate, reusable module and have moved the labelling of the benchmarks to two separate functions, so that the compiler can catch spelling mistakes for us (i.e. it prevents us from introducing a typo in the benchmark name).

Right now this will simply run the benchmarks every night at midnight but won't output any gnu plot.

### CODE REVIEW

**Comment by adinapoli-mndc** [comment]
This should now run at midnight and if I did instrument it correctly, should produce some artifacts we can look at later:

Screenshot 2019-09-09 at 16 52 37

**Comment by adinapoli-mndc** [comment]
@MeBrei After 1 day of running time, I have decided to kill the nightly job 😂 

https://buildkite.com/monadic/osrank-rs-nightly/builds/15#7f4408d8-aeac-430c-91ed-09bc0037f6bf

We definitely need to bump the parameters to make the run complete within this century 😬 . Feel free to tweak some knobs as part of this PR. All you have to do is to edit how the `cargo bench` is invoked as part of `.buildkite/nightly.yaml` :)

File: .buildkite/nightly.yaml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+steps:
+  - label: 'Benchmarks'
+    command: |
+        CRITERION_DEBUG=1 cargo bench -- --measurement-time 50 --sample-size 10 nightly
```
Comment (MeBrei) [inline .buildkite/nightly.yaml]: Tweaking the sample size here is redundant, as it is set directly in the benchmarks.

**Comment by MeBrei** [comment]
> We definitely need to bump the parameters to make the run complete within this century 😬 . Feel free to tweak some knobs as part of this PR. All you have to do is to edit how the cargo bench is invoked as part of .buildkite/nightly.yaml :)

Unfortunately this version of cargo does not allow for sample sizes below 10, so that I think we will have to limit the number of nodes/iterations directly in the code.

**Comment by adinapoli-mndc** [comment]
@MeBrei CI is currently funky but this is a red herring as the tests are just fine. Merging for now.

**Pull request merged**